pkg/auth/spire/certificate_provider_test.go

```diff
@@ -389,8 +390,24 @@ func TestSpireDelegateClient_GetCertificateForIdentity(t *testing.T) {
 				t.Errorf("SpireDelegateClient.GetCertificateForIdentity() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("SpireDelegateClient.GetCertificateForIdentity() = %v, want %v", got, tt.want)
+			if tt.want != nil {
+				if !assert.Equal(t, tt.want.Certificate, got.Certificate) {
+					t.Errorf("SpireDelegateClient.GetCertificateForIdentity().Certificate = %v, want %v", got.Certificate, tt.want.Certificate)
+				}
+				wantKey := tt.want.PrivateKey.(*rsa.PrivateKey)
+				gotKey := got.PrivateKey.(*rsa.PrivateKey)
+				if !wantKey.Equal(gotKey) {
+					if !assert.Equal(t, tt.want.PrivateKey, got.PrivateKey) {
+						t.Errorf("SpireDelegateClient.GetCertificateForIdentity().PrivateKey = %v, want %v", got.PrivateKey, tt.want.PrivateKey)
+					}
+				}
+				if !assert.Equal(t, tt.want.Leaf, got.Leaf) {
+					t.Errorf("SpireDelegateClient.GetCertificateForIdentity().Leaf = %v, want %v", got.Leaf, tt.want.Leaf)
+				}
+			} else {
+				if !assert.Equal(t, tt.want, got) {
```

Comment:
-> `assert.Nil(t, got, "SpireDelegateClient.GetCertificateForIdentity() = %v, should be nil")` ?